api/envoy/extensions/filters/network/thrift_proxy/v3/thrift_proxy.proto

```diff
@@ -82,6 +82,12 @@ message ThriftProxy {
   // compatibility, if no thrift_filters are specified, a default Thrift router filter
   // (`envoy.filters.thrift.router`) is used.
   repeated ThriftFilter thrift_filters = 5;
+
+  // If set to true, Envoy will try to skip decode data after metadata in the Thrift message.
+  // This mode will only work if the upstream and downstream protocols are the same and the transport
+  // is the same, the transport type is framed and the protocol is not Twitter. Otherwise Envoy will
+  // fallback to decode the data.
```

Comment:
Do you report out some stat to indicate if this fails or succeeds? Silent failure may be hard to diagnose.

Reply:
How about output some debug log，i that should be fine for diagnosis since thrift filter chain is unlikely to change dynamicly.

Reply:
Stat is also useful,  it's more intuitive, can I add it in another PR ?

Reply:
+1 to do it in a follow-up

Reply:
Shall we go ahead and land this so we can start testing it?

Reply:
It's fine in a followup PR if you are happy to own the potato (alternatively file an issue).